maestro-client/src/main/java/maestro/Maestro.kt

```diff
@@ -534,11 +505,66 @@ class Maestro(private val driver: Driver) : AutoCloseable {
         return driver.isUnicodeInputSupported()
     }
 
+    private fun waitUntilScreenIsStatic(timeoutMs: Long): Boolean {
+        if (isIOS) {
+            return MaestroTimer.retryUntilTrue(timeoutMs) {
+                val screenChanged = driver.isScreenChanged()
```

Comment:
It'd make sense to apply a heuristic matching algorithm when screenChanged returns false. This way we would have both the speedup of hash compare and the heuristic screen matching feature.